syntax = "proto3";

package analytics;

service AnalyticsService {
    rpc UpdateStatistics(UpdateStatisticsRequest) returns (StatisticsResponse);
    rpc GetMonthlyStatistics(GetMonthlyStatisticsRequest) returns (MonthlyStatisticsResponse);
}

message UpdateStatisticsRequest {
    string month_year = 1;
    int32 total_bookings_change = 2;
    int32 unique_users_change = 3;
}

message GetMonthlyStatisticsRequest {
    string month_year = 1;
}

message StatisticsResponse {
    bool success = 1;
    string message = 2;
}

message MonthlyStatistic {
    string month_year = 1;
    int32 total_bookings = 2;
    int32 unique_users = 3;
    string updated_at = 4;
}

message MonthlyStatisticsResponse {
    repeated MonthlyStatistic statistics = 1;
}